tools/autograph-monitor/monitor.go

```diff
@@ -113,6 +95,15 @@ func main() {
 	}
 }
 
+// Helper function to load a series of certificates and their hashes to a given truststore and hash list
+func LoadCertsToTruststore(hashArr *[]string, truststore *x509.CertPool, certificates []string) {
```

Comment:
This could create the `CertPool` inside itself and return the `hashArr` and the `CertPool`, couldn't it? Passing a pointer to a slice like this is not typical in Go, and hints that we may be too side-effect-y. (Even if we did take the slice as param, typical style is to return a slice -- that might be unchanged --  from the func instead to avoid `nil` pointer panics from being possible)

So the new type I'm proposing would be `loadCertsToTrustStore(certificates []string) ([]string, *x509.CertPool)`.

Reply:
omg duh

Reply:
Chaa, been there. Go (and Rust) are interesting because you get pointers (unlike most languages us web-y folks work in these days), but you also don't have to weird C stuff with them for simple things like output params etc etc!